libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/RegressionDistributionChart.tsx

```diff
@@ -0,0 +1,306 @@
+  getTheme
+} from "@fluentui/react";
+import {
+  BasicHighChart,
+  calculateBoxPlotDataFromErrorCohort,
+  defaultModelAssessmentContext,
+  ErrorCohort,
+  JointDataset,
+  ModelAssessmentContext
+} from "@responsible-ai/core-ui";
+import { localization } from "@responsible-ai/localization";
+import { PointOptionsObject } from "highcharts";
+import _ from "lodash";
+import React from "react";
+
+import { IBoxChartState } from "./BoxChartState";
+import { modelOverviewChartStyles } from "./ModelOverviewChart.styles";
+
+interface IRegressionDistributionChartProps {
+  cohorts: ErrorCohort[];
+  onChooseCohorts: () => void;
+  boxPlotState: IBoxChartState;
+  onBoxPlotStateUpdate: (boxPlotState: IBoxChartState) => void;
+}
+
+interface IRegressionDistributionChartState {
+  targetOption?: IChoiceGroupOption;
+  newlySelectedTargetOption?: IChoiceGroupOption;
+  targetFlyoutIsVisible: boolean;
+}
+
+const targetOptions: IChoiceGroupOption[] = [
+  {
+    key: JointDataset.PredictedYLabel,
+    text: localization.ModelAssessment.ModelOverview.regressionTargetOptions
+      .predictedY
+  },
+  {
+    key: JointDataset.TrueYLabel,
+    text: localization.ModelAssessment.ModelOverview.regressionTargetOptions
+      .trueY
+  },
+  {
+    key: JointDataset.RegressionError,
+    text: localization.ModelAssessment.ModelOverview.regressionTargetOptions
+      .error
+  }
+];
+
+export class RegressionDistributionChart extends React.Component<
+  IRegressionDistributionChartProps,
+  IRegressionDistributionChartState
+> {
+  public static contextType = ModelAssessmentContext;
+  public context: React.ContextType<typeof ModelAssessmentContext> =
+    defaultModelAssessmentContext;
+
+  public constructor(props: IRegressionDistributionChartProps) {
+    super(props);
+    this.state = { targetFlyoutIsVisible: false };
+  }
+
+  public componentDidMount(): void {
+    if (this.state.targetOption === undefined) {
+      const firstOption = targetOptions[0];
+      this.setState({
+        newlySelectedTargetOption: firstOption,
+        targetOption: firstOption
+      });
+    }
+  }
+
+  public async componentDidUpdate(
+    prevProps: IRegressionDistributionChartProps,
+    prevState: IRegressionDistributionChartState
+  ): Promise<void> {
+    if (
+      this.props.boxPlotState.boxPlotData.length === 0 ||
+      !_.isEqual(prevProps.cohorts, this.props.cohorts) ||
+      prevState.targetOption !== this.state.targetOption
+    ) {
+      const targetOption = this.state.targetOption
+        ? this.state.targetOption?.key.toString()
+        : targetOptions[0].key.toString();
+      const boxPlotData = await Promise.all(
+        this.props.cohorts.map((cohort: ErrorCohort, index: number) => {
+          return calculateBoxPlotDataFromErrorCohort(
+            cohort,
+            index,
+            targetOption
+          );
+        })
+      );
+      const outlierData = boxPlotData
+        .map((cohortBoxPlotData) => cohortBoxPlotData?.outliers)
```

Comment:
Is the logic to get outlier data is same in all places? can we have one common function to get the data if possible?

Reply:
I'm unifying it right now. The same can be done for the probability distribution chart for classification. I'm doing that as well.

Reply:
Done! This turned out to have a ton of potential to unify things. Both chart classes had the box and outlier calculation twice (although not identical, but upon investigating this in more detail it turned out to be easily rewriteable), so I was able to cut 75% of the associated logic 😮